### PR TITLE
OSSM-8296: Add a mTLS configuration doc

### PR DESCRIPTION
**Pull request by yxun**
#### What type of PR is this?

- [ ] Enhancement / New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [ ] Test
- [X] Documentation Update

#### What this PR does / why we need it:

This PR is part of OSSM-8296 Istio Security User Doc topics. It adds a new common doc file about configuring mTLS and concepts using OpenShift Service Mesh.
It also improves User doc explanations according to the existing OpenShift Service Mesh 2.x Security mTLS topic:
ref: https://docs.openshift.com/container-platform/4.17/service_mesh/v2x/ossm-security.html

### CODE REVIEW

**Comment by yxun** [comment]
following up : https://github.com/istio-ecosystem/sail-operator/pull/449

**Comment by longmuir** [comment]
This is great content - definitely needed in our product doc! @yxun, as there was a change vs 2.x where we had some settings in SMCP to enable mTLS, do you think it's worth adding a short paragraph toward the end of the [OSSM 2 vs 3 page](https://github.com/openshift-service-mesh/sail-operator/blob/main/docs/ossm/ossm2-vs-ossm3.md) to tell people that it is no longer possible, and you must follow the steps in this new doc page?

**Comment by yxun** [comment]
Thanks for review. I have added a paragraph toward the end of the [OSSM 2 vs 3 page] in this PR. 
I am checking the CI jobs now. Not clear why they were failed last week.

File: docs/ossm/ossm2-vs-ossm3.md [hunk 1]
```diff
@@ -98,4 +98,15 @@ By default, OpenShift Service Mesh 2 created Kubernetes `NetworkPolicy` resource
...
\ No newline at end of file
+OpenShift Service Mesh 3 does not create these policies, leaving it to the user to configure the level of isolation required for their environment. Istio provides fine grained access control of service mesh workloads through [Authorization Policies](https://istio.io/latest/docs/reference/config/security/authorization-policy/).
+
+## Service Mesh Security TLS Configuration
+
+By default, OpenShift Service Mesh 2 created the `ServiceMeshControlPlane` resource. You can enable mTLS strict mode by setting the `spec.security.dataPlane.mtls` to `true`. 
+You can set the minimum and maximum TLS protocol versions by setting the `spec.security.controlPlane.tls.minProtocolVersion` or `spec.security.controlPlane.tls.maxProtocolVersion` in your `ServiceMeshControlPlane` resource.
+
+OpenShift Service Mesh 3 does not create the `ServiceMeshControlPlane` resource. You can enable mTLS strict mode by applying the corresponding `PeerAuthentication` and `DestinationRule` resource(s). You can learn more about that in [Security mTLS Configuration](./security/security-mTLS-configuration.md).
```
Comment (longmuir) [inline docs/ossm/ossm2-vs-ossm3.md:108]:
```suggestion
In OpenShift Service Mesh 3, the `Istio` resource replaces the `ServiceMeshControlPlane` resource and does not include these settings. You can enable mTLS strict mode by applying the corresponding `PeerAuthentication` and `DestinationRule` resource(s). You can learn more about that in [Security mTLS Configuration](./security/security-mTLS-configuration.md).
```

**Review by longmuir — APPROVED** [review]

**Pull request merged**